### DISCUSSION TITLE
Customizing keypad SPC behavior

### DISCUSSION THREAD

**Discussion by nchisholm** [meow-edit/meow#413] [Q&A] [answered]
In keypad mode, you can type space to indicate that the following key is not modified by the control key.  This somewhat frustrates me for the `help-map` because the more common commands seem to be the unmodified ones.  For example, `C-h f` (SPC h SPC f) gives help on a callable while the shorter `C-h C-f` (SPC h f) sends you to Emacs FAQ, which I find myself wanting much less frequently.  The `ctl-x-map` also seems to put the more commonly used commands on bare keys.

Is there a way to have keypad mode add the ctrl modifier when you press space or otherwise send a bare key, reversing the current behavior?

Edit: Looks like someone asked something similar in a previous discussion.
https://github.com/meow-edit/meow/discussions/222

If the behavior is hardcoded and difficult to change, one solution might be to simply alter some keymaps to swap bare keys with modified ones.  I'll have to look into how to do that.

**Comment by DogLooksGood** [comment, marked as answer]
No, it's not possible.  The first SPC in `SPC h SPC f` stands for the command `meow-keypad` which bring you to keypad mode.  So, it is `h SPC f` for `C-h f`.  If you swap the behavior, it will be `SPC h f` for `C-h f`, then the whole key sequence is `SPC SPC h f`.  It's the same length.

**Reply by nchisholm** [reply]
Right - I see what you mean.  I'll try to write some function to "flip" a given keymap where C-[key] is replaced with [key] and vice versa.

**Reply by DogLooksGood** [reply]
If a command is frequently used and not easy to press, you can bind a key binding in leader keymap or `C-c <letter>`, make it `SPC <letter>`.  I believe we only have few commands like this.
